src/main/java/org/openstreetmap/atlas/geography/atlas/Atlas.java

```diff
@@ -764,31 +825,27 @@ default AtlasSize size()
 
     /**
      * Return a sub-atlas from this Atlas.
-     * <p>
-     * This would be a soft cut, meaning:
-     * <ul>
-     * <li>{@link Node}: It is included only if it is matched by the matcher, or if a valid edge
-     * (below) has it at one of its ends, or it is pulled in by an {@link Edge} which itself pulled
-     * in by a {@link Relation}, matched by the matcher.
-     * <li>{@link Edge}: It is included only if it is matched by the matcher or pulled in by a
-     * {@link Relation}, matched by the matcher.
-     * <li>{@link Area}: It is included only if it is matched by the matcher or pulled in by a
-     * {@link Relation}, matched by the matcher.
-     * <li>{@link Line}: It is included only if it is matched by the matcher or pulled in by a
-     * {@link Relation}, matched by the matcher.
-     * <li>{@link Point}: It is included only if it is matched by the matcher or pulled in by a
-     * {@link Relation}, matched by the matcher.
-     * <li>{@link Relation}: It is included if is matched by matcher or pulled in via another
-     * {@link Relation} which was matched by the matcher. To maintain {@link Relation} validity, all
-     * of its members will be included in the member list, even if not matched by the given matcher.
-     * </ul>
+     *
+     * @param boundary
+     *            The boundary within which the sub atlas will be built
+     * @param cutType
+     *            The type of cut to perform
+     * @return An optional sub-atlas. The optional will be empty in case there is nothing in the
+     *         {@link Polygon} after the cut was applied. Returning an empty atlas is not allowed.
+     */
+    Optional<Atlas> subAtlas(Polygon boundary, AtlasCutType cutType);
+
+    /**
+     * Return a sub-atlas from this Atlas.
      *
      * @param matcher
      *            The matcher to consider
-     * @return An optional sub-atlas. The optional will be empty in case the matcher would return an
-     *         empty atlas, which is not allowed.
+     * @param cutType
+     *            The type of cut to perform
+     * @return An optional sub-atlas. The optional will be empty in case the matcher and cut-type
+     *         return an empty atlas, which is not allowed.
      */
-    Optional<Atlas> subAtlas(Predicate<AtlasEntity> matcher);
+    Optional<Atlas> subAtlas(Predicate<AtlasEntity> matcher, AtlasCutType cutType);
```

Comment:
Please leave the backwards compatible function in the interface.

Reply:
Any algorithms operating on the Atlas interface won't be able to use the old function without casting or converting to a specific implementation.

Reply:
I'll leave it open for discussion, could go either why on this. Updating the API as a breaking change with the new signature makes sense to me, but if this causes too much churn, we can leave it as is, just not as clean, having two subAtlas methods here, with one doing the same thing as the other.

Reply:
I believe you can make a default implementation of the original signature that uses the newer signature. Then with you put a @Depricated on the old signature, and remove at the next major version increment. That would be my preference. Gives all the users a longer time to adjust for the same result.

Reply:
Good idea, I'll give that a shot.